src/main/java/io/quarkus/gizmo2/impl/Item.java

```diff
@@ -18,6 +18,7 @@
 import io.quarkus.gizmo2.impl.constant.ConstantImpl;
 
 public abstract non-sealed class Item implements Expr {
+    private final String creationSite = Util.callerOutsideGizmo();
```

Comment:
I wonder if this might have a significant performance impact. We might want to disable it by default.

Reply:
Right, I was expecting this comment :-) I have no idea how I would quantify the impact, but the number of `Item`s we allocate is nontrivial, so the impact here is likely to be nontrivial as well. I'll add a system property.